application/models/message.go

```diff
@@ -183,6 +185,90 @@ func (m *Message) Create(tx *pop.Connection, user User, requestUUID string, thre
 	return nil
 }
 
+// Todo Once gql is no longer supported, get rid of the Create() function above
+// CreateFromInput a new message if authorized.
+func (m *Message) CreateFromInput(tx *pop.Connection, user User, input api.MessageInput) *api.AppError {
+	var request Request
+	if err := request.FindByUUID(tx, input.RequestID); err != nil {
+		appError := api.AppError{
+			Category: api.CategoryUser,
+			Key:      api.MessageBadRequestUUID,
+			Err:      errors.New("error with new message: " + err.Error()),
+		}
+		return &appError
+	}
+
+	if isVisible, err := request.IsVisible(tx, user); err != nil {
+		appError := api.AppError{
+			Category: api.CategoryInternal,
+			Key:      api.QueryFailure,
+			Err:      errors.New("error with new message: " + err.Error()),
+		}
+		return &appError
+	} else if !isVisible {
+		appError := api.AppError{
+			Category: api.CategoryForbidden,
+			Key:      api.MessageRequestNotVisible,
+			Err:      errors.New("user cannot create a message on request"),
+		}
+		return &appError
+	}
+
+	var thread Thread
```

Comment:
Optional suggestion: wondering if this `if` block can/should be a separate function, if only to reduce the size of a long-ish function.